geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -108,21 +104,18 @@ public static int sdiffstore(RegionProvider regionProvider, RedisKey destination
   private static MemberSet calculateDiff(RegionProvider regionProvider, List<RedisKey> keys,
       boolean updateStats) {
     RedisSet firstSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(0), updateStats);
+    MemberSet diff = new MemberSet();
     if (firstSet.scard() == 0) {
-      return null;
+      return diff;
     }
-    MemberSet diff = new MemberSet(firstSet.members);
 
+    diff = new MemberSet(firstSet.members);
     for (int i = 1; i < keys.size(); i++) {
       RedisSet curSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(i), updateStats);
       if (curSet.scard() == 0) {
```

Comment:
Since we never store a empty RedisSet (if it is empty it just gets removed) scard will never return 0. You should delete this if block.